pystac/item.py

```diff
@@ -258,6 +259,31 @@ def from_dict(cls, d, href=None, root=None):
 
         return item
 
+    @property
+    def common_metadata(self):
+        """Access the item's common metadat fields as a CommonMetadata object
+
+        Returns:
+            CommonMetada: contains all common metadata fields in the items properties
+        """
+        return CommonMetadata(self.properties)
+
+    def set_common_metadata(self, common_metadata, override=False):
```

Comment:
Since properties edited through the common_metadata will edit the item's properties directly now, we should drop the `set_metadata` method.